### PR TITLE
Cypress/E2E: Fix E2E due to recent change in at-mention suggestion list

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Fix E2E due to [recent change in at-mention suggestion list](https://github.com/mattermost/mattermost-webapp/pull/7441) (with custom status), specifically related to the removal of [`.ml-2` class name](https://github.com/mattermost/mattermost-webapp/pull/7441/files#diff-38d76f42942a8534f255c0e3ee4e767dc70ff0083f309dfdbf47d5ebb7362cdcR176)

Added `cy.uiVerifyAtMentionInSuggestionList` and `cy. uiVerifyAtMentionSuggestion` to global custom commands to easily verify suggestion list, specifically for at-mention.

#### Ticket Link
none

### CODE REVIEW

**Comment by saturninoabril** [comment]
@calebroseland @prapti Gentle ping for your reviews. Just a bit blocking for the other PR. Thanks!

File: e2e/cypress/integration/account_settings/general/fullname_spec.js [hunk 1]
```diff
@@ -42,40 +42,36 @@ describe('Account Settings > Sidebar > General', () => {
...
 
     it('MM-T183 Filtering by first name with Korean characters', () => {
+        const {username} = testUser;
+
         cy.apiLogin(otherUser);
         cy.visit(`/${testTeam.name}/channels/town-square`);
 
-        // # type in user`s firstName substring
-        cy.get('#post_textbox').clear().type(`@정트리나${randomId}`);
+        // # Type in user`s firstName substring
```
Comment (prapti) [inline e2e/cypress/integration/account_settings/general/fullname_spec.js:58]:
```suggestion
        // # Type in user's first name substring
```

File: e2e/cypress/support/ui/suggestion_list.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+Cypress.Commands.add('uiVerifyAtMentionInSuggestionList', (sectionDividerName, user, isSelected = false) => {
+    // # Verify that the suggestion list is open and visible
```
Comment (prapti) [inline e2e/cypress/support/ui/suggestion_list.js:5]:
```suggestion
    // * Verify that the suggestion list is open and visible
```

File: e2e/cypress/support/ui/suggestion_list.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    return cy.get('#suggestionList').should('be.visible').within(() => {
+        if (sectionDividerName) {
+            // # Verify the section name is as expected
```
Comment (prapti) [inline e2e/cypress/support/ui/suggestion_list.js:8]:
```suggestion
            // * Verify the section name is as expected
```

File: e2e/cypress/support/ui/suggestion_list.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            cy.get('.suggestion-list__divider').findByText(sectionDividerName).should('be.visible');
+            cy.get('.suggestion-list__divider').next().findByTestId(`mentionSuggestion_${user.username}`).should('be.visible');
+        }
+
+        // # Verify that the user is selected
```
Comment (prapti) [inline e2e/cypress/support/ui/suggestion_list.js:13]:
```suggestion
        // * Verify that the user is selected
```

File: e2e/cypress/support/ui/suggestion_list.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+Cypress.Commands.add('uiVerifyAtMentionSuggestion', (user, isSelected = false) => {
+    const {
+        username,
+        first_name: firstName,
+        last_name: lastName,
+        nickname,
+    } = user;
+
+    // # Verify that the user is selected
```
Comment (prapti) [inline e2e/cypress/support/ui/suggestion_list.js:26]:
```suggestion
    // * Verify that the user is selected
```

**Review by prapti — APPROVED** [review]
Sorry, I only kept checking for the 'Dev Review' label on this PR in the github RHS. But thanks for the ping Saturn!

The updates look good to me. All tests passed. There are only minor changes needed in the comment.

Thank you for this update!

File: e2e/cypress/integration/account_settings/general/fullname_spec.js [hunk 6]
```diff
@@ -42,40 +42,36 @@ describe('Account Settings > Sidebar > General', () => {
             cy.get('#nameDesc').click();
 
             // * Set first name value
```
Comment (calebroseland) [inline e2e/cypress/integration/account_settings/general/fullname_spec.js:44]:
```suggestion
            // # Set first name value
```

File: e2e/cypress/integration/account_settings/general/fullname_spec.js [hunk 7]
```diff
@@ -42,40 +42,36 @@ describe('Account Settings > Sidebar > General', () => {
...
 
-        cy.findByTestId(testUser.username, {exact: false}).within((name) => {
-            cy.wrap(name).prev('.suggestion-list__divider').
-                should('have.text', 'Channel Members');
-            cy.wrap(name).find('.mention--align').
-                should('have.text', `@${testUser.username}`);
-            cy.wrap(name).find('.ml-2').
-                should('have.text', `정트리나${randomId}/trina.jung/집단사무국(CO) ${testUser.last_name} (${testUser.nickname})`);
-        });
+        // # Verify that the testUser is selected from mention autocomplete
```
Comment (calebroseland) [inline e2e/cypress/integration/account_settings/general/fullname_spec.js:61]:
```suggestion
        // * Verify that the testUser is selected from mention autocomplete
```

File: e2e/cypress/integration/account_settings/general/fullname_spec.js (hunk 6 above)
Comment (calebroseland) [inline e2e/cypress/integration/account_settings/general/fullname_spec.js:44]:
```suggestion
            // # Set first name value
```

File: e2e/cypress/integration/account_settings/general/fullname_spec.js [hunk 8]
```diff
@@ -42,40 +42,36 @@ describe('Account Settings > Sidebar > General', () => {
...
+        cy.uiVerifyAtMentionInSuggestionList('Channel Members', {...testUser, first_name: newFirstName}, true);
 
         // # Press tab on text input
         cy.get('#post_textbox').tab();
 
-        // # verify that after enter user`s username match
-        cy.get('#post_textbox').should('have.value', `@${testUser.username} `);
+        // # Verify that after enter user`s username match
```
Comment (calebroseland) [inline e2e/cypress/integration/account_settings/general/fullname_spec.js:67]:
```suggestion
        // * Verify that after enter user's username match
```

File: e2e/cypress/integration/account_settings/general/fullname_spec.js [hunk 9]
```diff
@@ -42,40 +42,36 @@ describe('Account Settings > Sidebar > General', () => {
...
+        cy.get('#post_textbox').should('have.value', `@${username} `);
 
-        // # click enter in post textbox
+        // # Click enter in post textbox
         cy.get('#post_textbox').type('{enter}');
 
-        // # verify that message has been post in chat
-        cy.get(`[data-mention="${testUser.username}"]`).
+        // # Verify that message has been post in chat
```
Comment (calebroseland) [inline e2e/cypress/integration/account_settings/general/fullname_spec.js:73]:
```suggestion
        // * Verify that message has been post in chat
```

File: e2e/cypress/integration/enterprise/elasticsearch_autocomplete/helpers/index.js [hunk 10]
```diff
@@ -238,9 +238,6 @@ module.exports = {
         cy.get('#suggestionList', {timeout: TIMEOUTS.FIVE_SEC}).should('be.visible');
 
         // # Verify user appears in results post-change
```
Comment (calebroseland) [inline e2e/cypress/integration/enterprise/elasticsearch_autocomplete/helpers/index.js:240]:
```suggestion
        // * Verify user appears in results post-change
```

File: e2e/cypress/integration/search_autocomplete/renaming_spec.js [hunk 11]
```diff
@@ -45,10 +45,7 @@ function searchAndVerifyUser(user) {
     cy.get('#suggestionList', {timeout: TIMEOUTS.FIVE_SEC}).should('be.visible');
 
     // # Verify user appears in results post-change
```
Comment (calebroseland) [inline e2e/cypress/integration/search_autocomplete/renaming_spec.js:47]:
```suggestion
    // * Verify user appears in results post-change
```

**Review by calebroseland — APPROVED** [review]
LGTM, suggested a few `#` vs `*`, but very minor, really.

**Comment by saturninoabril** [comment]
Thanks both for looking at those comment conventions! I didn't paid too much attention since they're there already but I must admit I should've done that.

**Pull request merged**